make/tests

```diff
@@ -12,8 +12,13 @@ GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST) -O$O
 $(GTEST)/src/gtest-all.o: CXXFLAGS += $(GTEST_CXXFLAGS)
 
 test/%$(EXE) : CXXFLAGS += $(GTEST_CXXFLAGS)
+ifneq (,$(findstring STAN_GPU, $(CXXFLAGS)))
```

Comment:
You said there was something weird where the `-lOpencl` has to come AFTER the .o specification and not before, only for GCC or something, right? We should first off make sure to document the exact issue (or put a link to a description) that in the relevant makefile. We can then also do this same behavior in a slightly prettier way - define a new flag, something like `POST_LDLIBS` and then you have something that looks like this:

```
POST_LDLIBS =
ifneq (,$(findstring STAN_GPU, $(CXXFLAGS)))
    POST_LDLIBS = -lOpenCL
endif
test/%$(EXE) : test/%.o $(GTEST_MAIN) $(GTEST)/src/gtest-all.o
	$(LINK.cpp) -o $@  $^ -lOpenCL
```

Reply:
Yeah that's a lot better. How do you feel about this [here](https://github.com/stan-dev/math/blob/2bd1c6bce78f85d0dc57a32ba0cd79803b235ebc/make/tests#L18)?